x/stakeibc/keeper/host_zone.go

```diff
@@ -50,22 +52,33 @@ func (k Keeper) GetHostZone(ctx sdk.Context, chain_id string) (val types.HostZon
 	return val, true
 }
 
-// GetHostZoneFromDenom returns a hostZone from the relevant coin denom
-
+// GetHostZoneFromDenom returns a hostZone from the relevant coin denom (denom is ibc/...)
 func (k Keeper) GetHostZoneFromDenom(ctx sdk.Context, denom string) (val types.HostZone, found bool) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.HostZoneKey))
 	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+	var bval types.HostZone
 
 	defer iterator.Close()
 
+	// hash is the part of denom after ibc/
+	hash := strings.Join(strings.SplitN(denom, "ibc/", -1), "")
+	req := &ibctypes.QueryDenomTraceRequest{Hash: hash}
+	goCtx := sdk.WrapSDKContext(ctx)
+	denomTrace, err := k.transferKeeper.DenomTrace(goCtx, req)
+	if err != nil {
+		k.Logger(ctx).Error("unable to obtain chain from denom %s: %w", hash, err)
+		return bval, false
+	}
+	baseDenom := strings.ToUpper(denomTrace.DenomTrace.BaseDenom)
```

Comment:
similar to above comment, I think it's worth adding an explicit check that the token has only traveled one hop, rather than implicitly verifying this with the loop below